src/utility/HCISharedMemTransport.cpp

```diff
@@ -338,8 +339,14 @@ static bool get_bd_address(uint8_t *bd_addr)
 
     bd_found = true;
   } else {
-    bd_addr = NULL;
-    bd_found = false;
+    uint8_t *otp_addr = OTP_Read(0);
+    if (otp_addr) {
+      memcpy(bd_addr, otp_addr, CONFIG_DATA_PUBADDR_LEN);
+      bd_found = true;
+    } else {
+      bd_addr = NULL;
```

Comment:
```suggestion
```
I would not set it to NULL as the call is:
`get_bd_address(bd_addr_udn)`
with `bd_addr_udn` :
`static uint8_t bd_addr_udn[CONFIG_DATA_PUBADDR_LEN];`

Reply:
yes, the boolean returned value is enough